src/main/java/core/basesyntax/strategy/SupplyHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class SupplyHandler implements OperationHandler {
+    private static final String SUPPLY = "s";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(SUPPLY)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
```

Comment:
The line 'listOfFruit.setQuantity(quantity + oldQuantity);' is updating the quantity of the 'listOfFruit' object, but 'oldQuantity' is always the quantity of the last 'fruitTransaction' processed, which might not be the correct logic. Verify the algorithm to ensure that the quantities are being aggregated correctly.